evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    any_values_are_nan,
+    keep_non_nan_rows,
+    pad_with_nans
+)
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    """Pipeline base class for time series regression problems."""
+
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series regression problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified as a dictionary with the "pipeline" key.
```

Comment:
"must be specified in a dictionary with the "pipeline" key"? Hmm. This was a tad bit confusing to me--not sure best way to clear it up via wording. Maybe would be good to give an example?

Reply:
Good suggestion - I added an example!